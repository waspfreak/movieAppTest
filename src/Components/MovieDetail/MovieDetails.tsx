import React from 'react';

// Components
import ProgressBar from '../ProgressBar/ProgressBar';

// Styled Components
import { StyledMovieDetails } from './style';

/**
 * Movie Detail UI component for user interaction
 */
export interface MovieDetailsProps {
	detailTitle: string,
	posterPath: string,
	modalOverview: string
	voteAverage: number,
}

export const MovieDetails: React.FC<MovieDetailsProps> = ({
	detailTitle,
	posterPath,
	modalOverview,
	voteAverage,
	...props
}) => {

	return (
		<StyledMovieDetails {...props}>
			<div className='movie_details-content'>
				<div className='movie_details-thumb'>
					<img className='image-thumb' src={posterPath} alt={detailTitle} />
				</div>
				<div className='movie_details-text'>
					<h3 className='movie_details_title'>{detailTitle}</h3>
					<p className='movie_details_description'>{modalOverview}</p>
					<div className='rating'>
						<div className='score'>
							<ProgressBar
								progress={voteAverage}
								size={50}
								strokeWidth={4}
								circleOneStroke='#3c0a0adb'
								circleTwoStroke='red'
							/>
						</div>
					</div>
				</div>
			</div>
		</StyledMovieDetails>
	);
};

export default MovieDetails;