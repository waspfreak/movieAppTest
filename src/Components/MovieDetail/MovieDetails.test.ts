import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {MovieDetails, MovieDetailsProps} from './MovieDetails';
const props: MovieDetailsProps[] = [
	{
	detailTitle: "Movie Title",
	posterPath: "/image/example.jpg",
	modalOverview: "Overview of movie",
	voteAverage: 55,
	},
];
snapshotTestFactory(MovieDetails, props);
