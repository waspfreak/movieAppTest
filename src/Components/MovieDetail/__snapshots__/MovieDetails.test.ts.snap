// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should match snapshot 1`] = `
.c1 {
  background: #111;
  border-radius: 50%;
  padding: 2px;
  display: inline-block;
}

.c2 {
  display: block;
  margin: 0px auto;
  max-width: 100%;
}

.c3 {
  fill: none;
}

.c4 {
  fill: none;
}

.c5 {
  font-family: 'Titillium Web',sans-serif;
  font-size: 14px;
  text-anchor: middle;
  fill: white;
  font-weight: bold;
  color: white;
}

.c0 {
  background-size: cover !important;
  background-position: center !important;
  width: 100%;
  padding: 40px 20px;
  box-sizing: border-box;
  -webkit-animation: animateMovieinfo 1s;
  animation: animateMovieinfo 1s;
}

.c0 .movie_details-content {
  max-width: 1280px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  background: rgb(0,0,0,0.8);
  position: relative;
}

.c0 .movie_details-thumb {
  margin-top: 20px;
}

.c0 .movie_details-thumb .image-thumb {
  width: 300px;
}

.c0 .movie_details-text {
  color: rgba(239,235,231,0.9);
  overflow: hidden;
  padding: 16px;
}

.c0 .movie_details-text .movie_details_title {
  font-size: 48px;
}

.c0 .movie_details-text .movie_details_description {
  font-size: 18px;
  line-height: 26px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.c0 .rating {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 .score {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 800;
  border-radius: 50%;
  margin: 0 0 0 0;
}

@media screen and (max-width:700px) {
  .c0 .movie_details-thumb .image-thumb {
    width: 100%;
    opacity: 0.5;
  }
}

@media screen and (max-width:700px) {
  .c0 .movie_details-text {
    position: absolute;
    margin-top: 25%;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    min-height: 600px;
    height: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="movie_details-content"
    >
      <div
        class="movie_details-thumb"
      >
        <img
          alt="Movie Title"
          class="image-thumb"
          src="/image/example.jpg"
        />
      </div>
      <div
        class="movie_details-text"
      >
        <h3
          class="movie_details_title"
        >
          Movie Title
        </h3>
        <p
          class="movie_details_description"
        >
          Overview of movie
        </p>
        <div
          class="rating"
        >
          <div
            class="score"
          >
            <div
              class="c1"
            >
              <svg
                class="c2"
                height="50"
                width="50"
              >
                <circle
                  class="c3"
                  cx="25"
                  cy="25"
                  r="23"
                  stroke="#3c0a0adb"
                  stroke-width="4"
                />
                <circle
                  class="c4"
                  cx="25"
                  cy="25"
                  r="23"
                  stroke="red"
                  stroke-dasharray="144.51326206513048"
                  stroke-dashoffset="65.03096792930872"
                  stroke-width="4"
                  style="transition: stroke-dashoffset 850ms ease-in-out;"
                />
                <text
                  class="c5"
                  x="25"
                  y="25"
                >
                  55
                  %
                </text>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
